sdk/lib/core/regexp.dart

```diff
@@ -82,22 +89,57 @@ abstract class RegExp implements Pattern {
   /// larger regular expression. Since a [String] is itself a [Pattern]
   /// which matches itself, converting the string to a regular expression
   /// isn't needed in order to search for just that string.
+  /// ```dart
+  /// print(RegExp.escape('dash@example.com')); // dash@example\.com
+  /// print(RegExp.escape('a+b')); // a\+b
+  /// print(RegExp.escape('a*b')); // a\*b
+  /// print(RegExp.escape('{a-b}')); // \{a-b\}
+  /// print(RegExp.escape('a?')); // a\?
+  /// ```
   external static String escape(String text);
 
   /// Finds the first match of the regular expression in the string [input].
   ///
   /// Returns `null` if there is no match.
+  /// ```dart
+  /// final string = '[00:13.37] This is a chat message.';
+  /// final regExp = RegExp(r'chat');
+  /// final match = regExp.firstMatch(string)!;
+  /// print(match[0]); // chat
+  /// ```
   RegExpMatch? firstMatch(String input);
 
   Iterable<RegExpMatch> allMatches(String input, [int start = 0]);
 
   /// Whether the regular expression has a match in the string [input].
+  /// ```dart
+  /// final string = 'Dash is a bird';
+  /// final regExp = RegExp(r'bird');
+  /// final match = regExp.hasMatch(string); // true
+  ///
+  /// final string = 'Dash is a bird';
+  /// final regExp = RegExp(r'dog');
+  /// final match = regExp.hasMatch(string); // false
```

Comment:
string, regExp and match should be var instead of final in first phase. later part is reusing those..